src/nb_clean/__init__.py

```diff
@@ -159,6 +165,7 @@ def check_notebook(
     notebook: nbformat.NotebookNode,
     *,
     remove_empty_cells: bool = False,
+    remove_all_notebook_metadata: bool = False,
```

Comment:
At the start of the function body, here and for all other functions that take both arguments, let's check that `remove_all_notebook_metadata` and `preserve_notebook_metadata` aren't both `True`, and raise an appropriate exception if so (this combination is impossible to satisfy).